bento/geometry/_geometry.py

```diff
@@ -1,310 +1,172 @@
-
-
-def rename_shapes(
-    data: AnnData,
-    mapping: Dict[str, str],
-    points_key: Optional[Union[List[str], None]] = None,
-    points_encoding: Union[List[Literal["label", "onehot"]], None] = None,
-    copy: bool = False,
-) -> Optional[AnnData]:
-    """Rename shape columns in adata.obs and points columns in adata.uns.
-
-    Parameters
-    ----------
-    adata : AnnData
-        Spatial formatted AnnData object
-    mapping : Dict[str, str]
-        Mapping of old shape names to new shape names
-    points_key : list of str, optional
-        List of keys for points DataFrame in `adata.uns`, by default None
-    points_encoding : {"label", "onehot"}, optional
-        Encoding type for each specified points
-    copy : bool, optional
-        Whether to return a copy of the AnnData object. Default False.
-
-    Returns
-    -------
-    AnnData
-        .obs: Updated shape column names
-        .uns[points_key]: Updated points shape(s) columns according to encoding type
-    """
-    adata = data.copy() if copy else data
-    adata.obs.rename(columns=mapping, inplace=True)
-
-    # Map point columns
-    if points_key:
-        # Get mapping for points column names
-        prefix_map = {
-            _get_shape_prefix(shape_name): _get_shape_prefix(new_name)
-            for shape_name, new_name in mapping.items()
-        }
-        # Get mapping for label encoding
-        label_map = {
-            re.sub(r"\D", "", shape_name): re.sub(r"\D", "", new_name)
-            for shape_name, new_name in prefix_map.items()
-        }
-
-        for p_key, p_encoding in zip(points_key, points_encoding):
-            if p_encoding == "label":
-                # Point column name with label encoding
-                col = re.sub(r"\d", "", list(prefix_map.keys())[0])
-                adata.uns[p_key][col] = adata.uns[p_key][col].astype(str).map(label_map)
-
-            elif p_encoding == "onehot":
-                # Remap column names
-                adata.uns[p_key].rename(columns=prefix_map, inplace=True)
-
-    return adata if copy else None
-
-
-def _get_shape_prefix(shape_name):
-    """Get prefix of shape name."""
-    return "_".join(shape_name.split("_")[:-1])
-
+    if sync and shape_name != "cell_boundaries":
+        if shape_name not in sdata.shapes["cell_boundaries"].columns:
+            raise ValueError(f"Shape {shape_name} not synced to cell_boundaries. Run sjoin_shapes first.")
+        return sdata.shapes["cell_boundaries"][shape_name]
+        
+    if shape_name not in sdata.shapes.keys():
+        raise ValueError(f"Shape {shape_name} not found in sdata.shapes")
+        
+    return sdata.shapes[shape_name].geometry
 
 def get_points(
-    data: AnnData, key: str = "points", asgeo: bool = False
-) -> Union[pd.DataFrame, gpd.GeoDataFrame]:
+    sdata: SpatialData, key: str = "transcripts", astype: str = "Pandas"
+) -> Union[pd.DataFrame, dd.DataFrame, gpd.GeoDataFrame]:
```

Comment:
- rename `astype` parameter to `type`
- use dataframe class types instead of `str` names

Reply:
Make Lowercase